Add support for traffic mirroring

wlix13 opened the pull request:
FEATURES:

* New Resource: aws_ec2_traffic_mirror_filter
* New Resource: aws_ec2_traffic_mirror_filter_rule
* New Resource: aws_ec2_traffic_mirror_session
* New Resource: aws_ec2_traffic_mirror_target

BUG FIXES:

* resource/aws_ec2_traffic_mirror_filter_rule: Fix bug in update function using incorrect field name

NOTES:

* acctest: Now uses "c2" partition for acceptance testing

wlix13 committed:
provider: add traffic mirroring resources

wlix13 committed:
website,mkdocs: add traffic mirroring pages

wlix13 committed:
acctest: update partition assignment logic

wlix13 committed:
ec2: correct rule number assignment

wlix13 committed:
ec2: update traffic mirror tests

- fix resource attributes check for filter rule test
- comment unsupported parameters for filter test
- convert target test to use network interface
- comment unsupported parameters for session test make it use network interface

wlix13 committed:
docs: update traffic mirroring docs

PixarV commented on internal/conns/config.go:
```diff
@@ -174,7 +174,11 @@ func (c *Config) Client(ctx context.Context) (interface{}, diag.Diagnostics) {
 			accountID = strings.Join([]string{projectName, customerName}, "@")
 		}
-		partition = "c2"
+		if p, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), c.Region); ok {
```
Лучше поправь partition для тестов: `Parition()` в acctest. Пусть он всегда возвращает "c2"

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+# Resource: aws_ec2_traffic_mirror_filter
+
+Provides an Traffic mirror filter.
+Read [limits and considerations](https://docs.aws.amazon.com/vpc/latest/mirroring/traffic-mirroring-considerations.html) for traffic mirroring
```
Ты тут по лимитам на использование ссылаешься на aws. Это ок?)

Еще мы немного иначе оформляем ссылки. Посмотри, например, в `ec2_transit_gateway.md`. Для ссылок ниже тоже поправь пожалуйста.

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+This resource supports the following arguments:
+
+* `description` - (Optional, Forces new resource) A description of the filter.
+* `network_services` - (Not supported) List of amazon network services that should be mirrored. Valid values: `amazon-dns`.
```
Посмотри в соседних файлах оформление неподдерживаемых аргументов

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+layout: "aws"
+page_title: "AWS: aws_ec2_traffic_mirror_filter"
+description: |-
+  Provides an Traffic mirror filter
```
```suggestion
  Provides a traffic mirror filter.
```

В описании также.

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
````diff
@@ -0,0 +1,55 @@
+```terraform
+import {
+  to = aws_ec2_traffic_mirror_filter.foo
+  id = "tmf-0fbb93ddf38198f64"
````
В качестве заглушки для id используется `tmf-12345678`

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
````diff
@@ -0,0 +1,55 @@
+Using `terraform import`, import traffic mirror filter using the `id`. For example:
+
+```console
+% terraform import aws_ec2_traffic_mirror_filter.foo tmf-0fbb93ddf38198f64
````
другой формат id

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+
+* `description` - (Optional, Forces new resource) A description of the filter.
+* `network_services` - (Not supported) List of amazon network services that should be mirrored. Valid values: `amazon-dns`.
+* `tags` - (Optional) Key-value map of resource tags. If configured with a provider [`default_tags` configuration block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
```
```suggestion
* `tags` - (Optional, Editable) Map of tags to assign to the traffic mirror filter. If a provider [`default_tags` configuration block][default-tags] is used, tags with matching keys will overwrite those defined at the provider level.
```

считай, утвержденная формулировка ) Editable еще можно докинуть, тк параметр редактируемый

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+
+## Attribute Reference
+
+This resource exports the following attributes in addition to the arguments above:
```
```suggestion
In addition to all arguments above, the following attributes are exported:
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+
+This resource exports the following attributes in addition to the arguments above:
+
+* `arn` - The ARN of the traffic mirror filter.
```
```suggestion
* `arn` - The Amazon Resource Name (ARN) of the traffic mirror filter.
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+This resource exports the following attributes in addition to the arguments above:
+
+* `arn` - The ARN of the traffic mirror filter.
+* `id` - The name of the filter.
```
В это поле не id приходит, а имя?

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,55 @@
+
+* `arn` - The ARN of the traffic mirror filter.
+* `id` - The name of the filter.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#default_tags-configuration-block).
```
```suggestion
* `tags_all` - Map of tags assigned to the the traffic mirror filter, including those inherited from the provider [`default_tags` configuration block][default-tags].
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -55,7 +55,7 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
 ## Argument Reference
 
-The following arguments are supported:
+This resource supports the following arguments:
```
Используем предыдущую формулировку

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -8,7 +8,7 @@ description: |-
 # Resource: aws_ec2_traffic_mirror_filter_rule
 
-Provides an Traffic mirror filter rule.  
+Provides an Traffic mirror filter rule.
```
```suggestion
Provides a traffic mirror filter rule.
```
и в description выше

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -73,17 +73,26 @@ Traffic mirror port range support following attributes:
+## Attribute Reference
 
-In addition to all arguments above, the following attributes are exported:
+This resource exports the following attributes in addition to the arguments above:
```
Используем предыдущую формулировку

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -55,7 +55,7 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
+This resource supports the following arguments:
 
 * `description` - (Optional) Description of the traffic mirror filter rule.
 * `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to which this rule should be added
```
Первыми в списке идут обязательные по алфавиту, далее опциональные по алфавиту

В конце некоторых описаний не хватает точек.

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
````diff
@@ -73,17 +73,26 @@ Traffic mirror port range support following attributes:
+```terraform
+import {
+  to = aws_ec2_traffic_mirror_filter_rule.rule
+  id = "tmf-0fbb93ddf38198f64:tmfr-05a458f06445d0aee"
````
Поправить формат id и ниже

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -73,17 +73,26 @@ Traffic mirror port range support following attributes:
-In addition to all arguments above, the following attributes are exported:
+This resource exports the following attributes in addition to the arguments above:
 
 * `arn` - ARN of the traffic mirror filter rule.
```
```suggestion
* `arn` - The Amazon Resource Name (ARN) of the traffic mirror filter rule.
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -73,17 +73,26 @@ Traffic mirror port range support following attributes:
+This resource exports the following attributes in addition to the arguments above:
 
 * `arn` - ARN of the traffic mirror filter rule.
 * `id` - Name of the traffic mirror filter rule.
```
Тут тоже по-моему id, а не name

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
Те же замечания, что и к файлам выше

PixarV commented on website/docs/r/ec2_traffic_mirror_target.md:
Те же замечания, что и к файлам выше

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -17,17 +17,16 @@ To create a basic traffic mirror session
 
 resource "aws_ec2_traffic_mirror_target" "target" {
-  network_load_balancer_arn = aws_lb.lb.arn
+  network_interface_id = aws_instance.dst.primary_network_interface_id
```
Если ссылаешься на сущность, в идеале ее бы в этом примере создать

PixarV replied:
Можно сделать, например, как в примере для сессии

PixarV commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,59 @@
+
+This resource exports the following attributes in addition to the arguments above:
+
+* `id` - The ID of the Traffic Mirror target.
```
```suggestion
* `id` - The ID of the traffic mirror target.
```

ресурсы с маленькой буквы

PixarV commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,59 @@
+* `id` - The ID of the Traffic Mirror target.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#default_tags-configuration-block).
+* `arn` - The ARN of the traffic mirror target.
+* `owner_id` - The ID of the AWS account that owns the traffic mirror target.
```
ID проекта

PixarV commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,59 @@
+
+* `id` - The ID of the Traffic Mirror target.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#default_tags-configuration-block).
+* `arn` - The ARN of the traffic mirror target.
```
отсортируй по алфавиту пожалуйста

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -36,30 +35,39 @@ resource "aws_ec2_traffic_mirror_session" "session" {
+This resource supports the following arguments:
 
 * `description` - (Optional) A description of the traffic mirror session.
 * `network_interface_id` - (Required, Forces new) ID of the source network interface. Not all network interfaces are eligible as mirror sources. On EC2 instances only nitro based instances support mirroring.
```
nitro based instances?

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -36,30 +35,39 @@ resource "aws_ec2_traffic_mirror_session" "session" {
 * `description` - (Optional) A description of the traffic mirror session.
 * `network_interface_id` - (Required, Forces new) ID of the source network interface. Not all network interfaces are eligible as mirror sources. On EC2 instances only nitro based instances support mirroring.
 * `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to be used
 * `traffic_mirror_target_id` - (Required) ID of the traffic mirror target to be used
```
```suggestion
* `traffic_mirror_filter_id`  - (Required) The ID of the traffic mirror filter to be used.
* `traffic_mirror_target_id` - (Required) The ID of the traffic mirror target to be used.
.```

И обязательные поля перенеси выше опциональных

PixarV reviewed (changes requested):
Только дока.
\+ линтер упал, поправь пожалуйста.

Diyckstra commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+
+The following arguments are supported:
+
+* `description` - (Optional, Forces new resource) A description of the filter.
```
В `ec2_traffic_mirror_target.md`  написано просто `Forces new` или там добавить ресурс или тут убрать

Diyckstra commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,44 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
 
 The following arguments are supported:
```
Сюда добавить что-то вроде:
`All arguments except "traffic_direction" are editable.`

DiGa66 replied:
тогда  \`traffic_direction\`

Diyckstra commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,73 @@
+* `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to be used.
+* `traffic_mirror_target_id` - (Required) ID of the traffic mirror target to be used.
+* `description` - (Optional) A description of the traffic mirror session.
+* `tags` - (Optional, Editable) Map of tags to assign to the traffic mirror session. If a provider [`default_tags` configuration block][default-tags] is used, tags with matching keys will overwrite those defined at the provider level.
```
В раздеплоенной доке это выглядит странно

PixarV replied:
Нужно добавить определение для метки `default-tags` 

```
[default-tags]: https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block
[traffic-mirroring]: https://docs.k2.cloud/en/services/interconnect/traffic_mirroring.html

# Resource: aws_ec2_traffic_mirror_session
 ...
```

Diyckstra commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,73 @@
+* `arn` - The Amazon Resource Name (ARN) of the traffic mirror session.
+* `id` - The ID of the traffic mirror session.
+* `owner_id` - The ID of the project that owns the traffic mirror session.
+* `tags_all` - Map of tags assigned to the the traffic mirror session, including those inherited from the provider [`default_tags` configuration block][default-tags].
```
В раздеплоенной доке это выглядит странно

PixarV replied:
```suggestion
* `tags_all` - Map of tags assigned to the traffic mirror session, including those inherited from the provider [`default_tags` configuration block][default-tags].
```

wlix13 committed:
ec2,docs: comment out unsupported parameters in traffic mirror resources

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
 
 Traffic mirror port range support following attributes:
 
 * `from_port` - (Optional) Starting port of the range
```
В коде есть валидация, что значения от 0 до 65535. Нужно добавить про это информацию в доку.

В пользовательской доке значения стартуют с 1, а не с 0 ([link](https://docs.k2.cloud/ru/api/ec2/datatypes/TrafficMirrorPortRange.html#trafficmirrorportrange)) Нужно поправить валидацию?

wlix13 replied:
В доке неправильно оказалось, API позволяет сделать с 0, фикс в доку уже есть

PixarV commented on internal/service/ec2/vpc_traffic_mirror_session_test.go:
```diff
@@ -20,11 +20,11 @@ import (
-	description := "test session"
-	session := sdkacctest.RandIntRange(1, 32766)
+	// description := "test session"
+	session := sdkacctest.RandIntRange(1, 128)
```
Поправь максимальное значение в валидации у параметра `session_number`

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+layout: "aws"
+page_title: "AWS: aws_ec2_traffic_mirror_filter"
+description: |-
+  Provides a traffic mirror filter
```
```suggestion
  Provides a traffic mirror filter.
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+# Resource: aws_ec2_traffic_mirror_filter
+
+Provides a traffic mirror filter.
+[traffic-mirroring]: https://docs.k2.cloud/en/services/interconnect/traffic_mirroring.html
```
Все определения ссылок перемести пожалуйста над `# Resource: ...` и отсортируй
```
[default-tags]: ...
[import]: ...
[traffic-mirroring]: ...

# Resource: aws_ec2_traffic_mirror_filter
```
На остальных страничках тоже поправь пожалуйста

PixarV replied:
`[traffic-mirroring]` хорошо бы еще где-нибудь использовать. Обычно формат такой:

```
Provides a traffic mirror filter. For details about traffic mirroring, see the [user documentation][traffic-mirroring].
```

на всех страничках

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
 
 Traffic mirror port range support following attributes:
 
 * `from_port` - (Optional) Starting port of the range
```
Точка в конце + `to_port`

PixarV commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,82 @@
+* `arn` - The Amazon Resource Name (ARN) of the traffic mirror target.
+* `id` - The ID of the traffic mirror target.
+* `owner_id` - The ID of the project that owns the traffic mirror target.
+* `tags_all` - Map of tags assigned to the the traffic mirror target, including those inherited from the provider [`default_tags` configuration block][default-tags].
```
```suggestion
* `tags_all` - Map of tags assigned to the traffic mirror target, including those inherited from the provider [`default_tags` configuration block][default-tags].
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
+* `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to which this rule should be added.
+* `description` - (Optional) Description of the traffic mirror filter rule.
+* `destination_port_range` - (Optional) Destination port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below.
+* `protocol` - (Optional) Protocol number, for example 17 (UDP), to assign to the Traffic Mirror rule. For information about the protocol value, see [Protocol Numbers](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml) on the Internet Assigned Numbers Authority (IANA) website.
```
```suggestion
// в блоке со ссылками наверху
[Protocol Numbers]: https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml
...

* `protocol` - (Optional) Protocol number, for example 17 (UDP), to assign to the traffic mirror rule. For information about the protocol value, see [Protocol Numbers] on the Internet Assigned Numbers Authority (IANA) website.
```

С маленькой буквы Traffic Mirror + оформление ссылки

DiGa66 replied:
```suggestion
* `protocol` - (Optional) Protocol number, for example, 17 (UDP), to assign to the traffic mirror rule. For information about the protocol value, see [Protocol Numbers](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml) on the Internet Assigned Numbers Authority (IANA) website.
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
 
-* `description` - (Optional) Description of the traffic mirror filter rule.
-* `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to which this rule should be added
 * `destination_cidr_block` - (Required) Destination CIDR block to assign to the Traffic Mirror rule.
```
Traffic Mirror с маленькой буквы, ниже тоже поправь пожалуйста

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
 * `to_port` - (Optional) Ending port of the range
 
-## Attributes Reference
+All arguments except "traffic_direction" are editable.
```
В схеме для `traffic_direction` не указано, что параметр `ForceNew: true`, судя по [пользовательской доке](https://docs.k2.cloud/ru/api/ec2/traffic_mirroring/ModifyTrafficMirrorFilterRule.html) этот параметр тоже можно обновлять. А вот `traffic_mirror_filter_id` нельзя. 

 Можно писать что-то типа 
```
All arguments except for `traffic_mirror_filter_id` are editable.
```
или можно докинуть Editable ко всем редактируемым параметрам, например, `(Optional, Editable)`

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
+* `protocol` - (Optional) Protocol number, for example 17 (UDP), to assign to the Traffic Mirror rule. For information about the protocol value, see [Protocol Numbers](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml) on the Internet Assigned Numbers Authority (IANA) website.
+* `source_port_range` - (Optional) Source port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below.
 
 Traffic mirror port range support following attributes:
```
Блоки удобно оформлять подсекцией и потом на нее ссылаться:

```
* `destination_port_range` - (Optional) Destination port range. Supported only when the protocol is set to TCP(6) or UDP(17). The structure of this block is [described below](#traffic-mirror-port-range).
* `protocol` - ...
* `source_port_range` - (Optional) Source port range. Supported only when the protocol is set to TCP(6) or UDP(17). The structure of this block is [described below](#traffic-mirror-port-range).

### Traffic mirror port range

The block has the following structure:

* `from_port` - (Optional, Editable) Starting port of the range.
* `to_port` - (Optional, Editable) Ending port of the range.
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
 * `source_cidr_block` - (Required) Source CIDR block to assign to the Traffic Mirror rule.
-* `source_port_range` - (Optional) Source port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below
-* `traffic_direction` - (Required) Direction of traffic to be captured. Valid values are `ingress` and `egress`
+* `traffic_direction` - (Required) Direction of traffic to be captured. Valid values are `ingress` and `egress`.
```
```suggestion
* `traffic_direction` - (Required) Direction of traffic to be captured.
    * _Valid values:_ `ingress`, `egress`
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
-* `destination_port_range` - (Optional) Destination port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below
-* `protocol` - (Optional) Protocol number, for example 17 (UDP), to assign to the Traffic Mirror rule. For information about the protocol value, see [Protocol Numbers](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml) on the Internet Assigned Numbers Authority (IANA) website.
-* `rule_action` - (Required) Action to take (accept | reject) on the filtered traffic. Valid values are `accept` and `reject`
+* `rule_action` - (Required) Action to take (accept | reject) on the filtered traffic. Valid values are `accept` and `reject`.
```
```suggestion
* `rule_action` - (Required) Action to take (accept | reject) on the filtered traffic.
    * _Valid values:_ `accept`, `reject`
```

DiGa66 replied:
а нужно ли это уточнение в скобках, если ниже действия конкретизируются в valid values?

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
-* `protocol` - (Optional) Protocol number, for example 17 (UDP), to assign to the Traffic Mirror rule. For information about the protocol value, see [Protocol Numbers](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml) on the Internet Assigned Numbers Authority (IANA) website.
-* `rule_action` - (Required) Action to take (accept | reject) on the filtered traffic. Valid values are `accept` and `reject`
+* `rule_action` - (Required) Action to take (accept | reject) on the filtered traffic. Valid values are `accept` and `reject`.
 * `rule_number` - (Required) Number of the Traffic Mirror rule. This number must be unique for each Traffic Mirror rule in a given direction. The rules are processed in ascending order by rule number.
```
Можно докинуть про ограничение значения от 1 до 128, и в схему в виде валидации тоже.

```
* `rule_number` - ...
    * _Valid values:_ From 1 to 128
```

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
-* `source_port_range` - (Optional) Source port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below
-* `traffic_direction` - (Required) Direction of traffic to be captured. Valid values are `ingress` and `egress`
+* `traffic_direction` - (Required) Direction of traffic to be captured. Valid values are `ingress` and `egress`.
+* `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to which this rule should be added.
```
The ID of

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
````diff
@@ -0,0 +1,73 @@
+```terraform
+import {
+  to = aws_ec2_traffic_mirror_session.session
+  id = "tms-123456778"
````
```suggestion
  id = "tms-12345678"
```

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
````diff
@@ -0,0 +1,73 @@
+Using `terraform import`, import traffic mirror session using the `id`. For example:
+
+```console
+% terraform import aws_ec2_traffic_mirror_session.session tms-123456778
````
```suggestion
% terraform import aws_ec2_traffic_mirror_session.session tms-12345678
```

PixarV commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,82 @@
+The following arguments are supported:
+
+* `description` - (Optional, Forces new) A description of the traffic mirror target.
+* `network_interface_id` - (Required, Forces new) The network interface ID that is associated with the target.
```
Обязательные поля выше опциональных

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+
+# Resource: aws_ec2_traffic_mirror_filter
+
+Provides a traffic mirror filter.
```
Provides -> Manages

На всех страничках

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,73 @@
+
+The following arguments are supported:
+
+* `network_interface_id` - (Required, Forces new) ID of the source network interface.
```
К остальным параметрам тоже докинь `Forces new` или `Editable` в зависимости от того поддерживают ли они редактирование

DiGa66 replied:
Forces new resource (так в других местах оформлено)

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,73 @@
+
+* `network_interface_id` - (Required, Forces new) ID of the source network interface.
+* `session_number` - (Required) - The session number determines the order in which sessions are evaluated when an interface is used by multiple sessions. The first session with a matching filter is the one that mirrors the packets.
+* `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to be used.
```
Лишние пробелы перед `-`

PixarV commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,73 @@
+The following arguments are supported:
+
+* `network_interface_id` - (Required, Forces new) ID of the source network interface.
+* `session_number` - (Required) - The session number determines the order in which sessions are evaluated when an interface is used by multiple sessions. The first session with a matching filter is the one that mirrors the packets.
```
```suggestion
* `session_number` - (Required) The session number determines the order in which sessions are evaluated when an interface is used by multiple sessions. The first session with a matching filter is the one that mirrors the packets.
```

PixarV reviewed (changes requested):
больше доки

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+
+## Example Usage
+
+To create a basic traffic mirror filter
```
я бы вообще убрал, обычно в примерах использования обходимся без описаний, либо так:
(аналогично сделать ниже)
```suggestion
To create a basic traffic mirror filter, use:
```

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+
+The following arguments are supported:
+
+* `description` - (Optional, Forces new) A description of the filter.
```
В других местах в такой формулировке, поменять везде ниже
У нас обычно (The) description (либо определенный артикль, либо без артикля.
Давайте писать без артикля, как в основном в AWS
```suggestion
* `description` - (Optional, Forces new resource) Description of the filter.
```

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter.md:
````diff
@@ -0,0 +1,56 @@
+}
+```
+
+Using `terraform import`, import traffic mirror filter using the `id`. For example:
````
А как это связано с предыдущим -- это продолжение или противопоставление? Во втором случае здесь не надо добавить что-то вроде: 
In older Terraform versions, ...
У AWS как-то очень коряво написано
```suggestion
In older Terraform versions, the traffic mirror filter can be imported by its `id` using `terraform import`, e.g.:
```

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+
+## Import
+
+In Terraform v1.5.0 and later, use an `import` block to import traffic mirror filter using the `id`.
```
```suggestion
In Terraform v1.5.0 or later, traffic mirror filter can be imported by `id` using the `import` block.
```

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -3,22 +3,21 @@ subcategory: "VPC (Virtual Private Cloud)"
 ## Example Usage
 
-To create a basic traffic mirror session
+To create a basic traffic mirror filter rule
```
сделать по аналогии с комментом про фильтр

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
+* `traffic_direction` - (Required) Direction of traffic to be captured. Valid values are `ingress` and `egress`.
+* `traffic_mirror_filter_id`  - (Required) ID of the traffic mirror filter to which this rule should be added.
+* `description` - (Optional) Description of the traffic mirror filter rule.
+* `destination_port_range` - (Optional) Destination port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below.
```
сделать ссылкой на соответствующий блок Traffic mirror port range

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -57,33 +56,46 @@ resource "aws_ec2_traffic_mirror_filter_rule" "rulein" {
+* `description` - (Optional) Description of the traffic mirror filter rule.
+* `destination_port_range` - (Optional) Destination port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below.
+* `protocol` - (Optional) Protocol number, for example 17 (UDP), to assign to the Traffic Mirror rule. For information about the protocol value, see [Protocol Numbers](https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml) on the Internet Assigned Numbers Authority (IANA) website.
+* `source_port_range` - (Optional) Source port range. Supported only when the protocol is set to TCP(6) or UDP(17). See Traffic mirror port range documented below.
```
Ссылка на блок Traffic mirror port range

DiGa66 commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,73 @@
+
+## Example Usage
+
+To create a basic traffic mirror session
```
оформление по аналогии

DiGa66 commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,82 @@
+
+## Example Usage
+
+To create a basic traffic mirror target
```
оформление как выше

DiGa66 commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,82 @@
+* `owner_id` - The ID of the project that owns the traffic mirror target.
+* `tags_all` - Map of tags assigned to the the traffic mirror target, including those inherited from the provider [`default_tags` configuration block][default-tags].
+
+## Import
```
как выше

PixarV commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -0,0 +1,108 @@
+* `to_port` - (Optional, Editable) Ending port of the range.
+    * _Valid values:_ From 0 to 65535
+
+All arguments except for `traffic_mirror_filter_id` are editable.
```
Строчку можно дропнуть, тк выше для всех редактируемых аргументов это явно указано.

PixarV commented on website/docs/r/ec2_traffic_mirror_filter.md:
```diff
@@ -0,0 +1,56 @@
+---
+subcategory: "VPC (Virtual Private Cloud)"
+layout: "aws"
+page_title: "AWS: aws_ec2_traffic_mirror_filter"
```
```suggestion
page_title: "aws_ec2_traffic_mirror_filter"
```

На других страничках тоже убери, пожалуйста )

wlix13 committed:
acctest: update PreCheck to include env vars

wlix13 committed:
ec2: update traffic mirror tests to use ARN matching without account

- traffic mirror filter don't return owner_id from API
- traffic mirror filter rule don't return owner_id from API

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -0,0 +1,106 @@
+The following arguments are supported:
+
+* `destination_cidr_block` - (Required, Editable) Destination CIDR block to assign to the traffic mirror rule.
+* `rule_action` - (Required, Editable) Action to take (accept | reject) on the filtered traffic.
```
я бы убрал значение действий -- они в valid values указаны
тогда ниже везде поменять
```suggestion
* `rule_action` - (Required, Editable) Action to take on the filtered traffic.
```

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
````diff
@@ -0,0 +1,106 @@
+}
+```
+
+Using `terraform import`, import traffic mirror filter rule using the `traffic_mirror_filter_id` and `id` separated by `:`. For example:
````
аналогично оформлению выше
In older Terraform versions, the traffic mirror filter rule can be imported by `traffic_mirror_filter_id` and its `id`  separated by `:` using `terraform import`, e.g.:

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -0,0 +1,106 @@
+* `source_cidr_block` - (Required, Editable) Source CIDR block to assign to the traffic mirror rule.
+* `traffic_direction` - (Required, Editable) Direction of traffic to be captured.
+    * _Valid values:_ `ingress`, `egress`
+* `traffic_mirror_filter_id` - (Required) The ID of the traffic mirror filter to which this rule should be added.
```
В оригинальной документации здесь артикля нет, но в других аналогичных местах есть (как и у нас).
При описании других аргументов тоже обычно используется определённый артикль и у нас и у AWS, но здесь в оригинале все описания аргументов без артикля.
В общем я бы тогда и здесь пока убрал, чтобы все описания были единообразны.

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
```diff
@@ -0,0 +1,106 @@
+* `protocol` - (Optional, Editable) Protocol number, for example 17 (UDP), to assign to the traffic mirror rule. For information about the protocol value, see [Protocol Numbers][protocol-numbers] on the Internet Assigned Numbers Authority (IANA) website.
+* `source_port_range` - (Optional, Editable) Source port range. Supported only when the protocol is set to TCP(6) or UDP(17). The structure of this block is [described below](#traffic-mirror-port-range).
+
+### Traffic mirror port range
```
@PixarV Мы используем заголовка название аргумента/атрибута, но такого раньше вроде не встречалось -- один блок для 2 разных аргументов. Так оставляем?

DiGa66 commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,103 @@
+* `session_number` - (Required, Editable) The session number determines the order in which sessions are evaluated when an interface is used by multiple sessions. The first session with a matching filter is the one that mirrors the packets.
+* `traffic_mirror_filter_id` - (Required, Editable) ID of the traffic mirror filter to be used.
+* `traffic_mirror_target_id` - (Required, Editable) ID of the traffic mirror target to be used.
+* `description` - (Optional, Editable) A description of the traffic mirror session.
```
```suggestion
* `description` - (Optional, Editable) Description of the traffic mirror session.
```

DiGa66 commented on website/docs/r/ec2_traffic_mirror_session.md:
````diff
@@ -0,0 +1,103 @@
+}
+```
+
+Using `terraform import`, import traffic mirror session using the `id`. For example:
````
оформление

DiGa66 commented on website/docs/r/ec2_traffic_mirror_target.md:
```diff
@@ -0,0 +1,82 @@
+The following arguments are supported:
+
+* `network_interface_id` - (Required, Forces new) The network interface ID that is associated with the target.
+* `description` - (Optional, Forces new) A description of the traffic mirror target.
```
оформление

DiGa66 commented on website/docs/r/ec2_traffic_mirror_target.md:
````diff
@@ -0,0 +1,82 @@
+}
+```
+
+Using `terraform import`, import traffic mirror target using the `id`. For example:
````
оформление

DiGa66 commented on website/docs/r/ec2_traffic_mirror_filter_rule.md:
````diff
@@ -0,0 +1,106 @@
+}
+```
+
+In older Terraform versions, the traffic mirror filter rule can be imported by traffic_mirror_filter_id and its id separated by : using terraform import, e.g.:
````
```suggestion
In older Terraform versions, the traffic mirror filter rule can be imported by `traffic_mirror_filter_id` and its `id` separated by `:` using `terraform import`, e.g.:
```

DiGa66 commented on website/docs/r/ec2_traffic_mirror_session.md:
```diff
@@ -0,0 +1,103 @@
+The following arguments are supported:
+
+* `network_interface_id` - (Required, Forces new resource) ID of the source network interface.
+* `session_number` - (Required, Editable) The session number determines the order in which sessions are evaluated when an interface is used by multiple sessions. The first session with a matching filter is the one that mirrors the packets.
```
```suggestion
* `session_number` - (Required, Editable) The session number determines the order in which sessions are evaluated when the interface is used by multiple sessions. The first session with a matching filter is the one that mirrors the packets.
```
